src/librustc_codegen_llvm/builder.rs

```diff
@@ -652,6 +652,56 @@ impl BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
         unsafe { llvm::LLVMBuildSExt(self.llbuilder, val, dest_ty, UNNAMED) }
     }
 
+    fn fptoui_sat(&mut self, val: &'ll Value, dest_ty: &'ll Type) -> Option<&'ll Value> {
+        if self.sess().target.target.arch == "wasm32"
```

Comment:
Oh as another note, could this have a comment about why there's an `if` block here? Basically saying that wasm has a native instruction which exactly matches Rust's saturating semantics, except that it's still behind a target feature so we need to test for that.